main.cpp

```diff
@@ -1,7 +1,314 @@
+    int start_1, end_1, start_2, end_2 = 0;
+    for (int i=0;i<orderedTask.size();i++){    
+        if(lastMaintenance(machine1Blocks) == maxTimeWork)
+            addBlock(machine1Blocks, getCmax(machine1Blocks), getCmax(machine1Blocks) + maintenance, "maintenance");
+        if(lastMaintenance(machine2Blocks) == maxTimeWork)
+            addBlock(machine2Blocks, getCmax(machine2Blocks), getCmax(machine2Blocks) + maintenance, "maintenance");
+        int expectedMachine1End = fitTaskOnMachine(machine1Blocks, maintenance, tasks[orderedTask[i]].get_op1(), maxTimeWork, latency);
+        int noWaitBreak = 0;
+        if (expectedMachine1End < getCmax(machine2Blocks)){
+            noWaitBreak = getCmax(machine2Blocks) - expectedMachine1End;
+            if (noWaitBreak >= maintenance){
+                addBlock(machine2Blocks, getCmax(machine1Blocks), getCmax(machine1Blocks) + maintenance, "maintenance");
+                noWaitBreak = noWaitBreak - maintenance;
+            }
+            addBlock(machine1Blocks, getCmax(machine1Blocks) + noWaitBreak, getCmax(machine1Blocks) + noWaitBreak + tasks[orderedTask[i]].get_op1(), "task", tasks[orderedTask[i]].get_task_number(), 1, maintenance,tasks[orderedTask[i]].get_op1(), maxTimeWork, latency);
+            addBlock(machine2Blocks, getCmax(machine1Blocks), getCmax(machine1Blocks) + tasks[orderedTask[i]].get_op2(), "task", tasks[orderedTask[i]].get_task_number(), 2, maintenance, tasks[orderedTask[i]].get_op2(), maxTimeWork, latency);
+        }
+        else{
+            noWaitBreak = expectedMachine1End - getCmax(machine2Blocks);
+            if (noWaitBreak >= maintenance){
+                addBlock(machine2Blocks, getCmax(machine2Blocks), getCmax(machine2Blocks) + maintenance, "maintenance");
+            }
+            addBlock(machine1Blocks, getCmax(machine1Blocks), getCmax(machine1Blocks) + tasks[orderedTask[i]].get_op1(), "task", tasks[orderedTask[i]].get_task_number(), 1, maintenance,tasks[orderedTask[i]].get_op1(), maxTimeWork, latency);
+            addBlock(machine2Blocks, getCmax(machine1Blocks), getCmax(machine1Blocks) + tasks[orderedTask[i]].get_op2(), "task", tasks[orderedTask[i]].get_task_number(), 2, maintenance,tasks[orderedTask[i]].get_op2(), maxTimeWork, latency);
+        }
+    }
+}
+
+
+std::vector <int> randomTasksOrder(const std::vector <Task> &tasks){
+    std::vector <int> randomOrder;
+    for (int i=0; i<tasks.size(); i++)
+        randomOrder.push_back(i);
+    random_shuffle(randomOrder.begin(), randomOrder.end()); 
+    return randomOrder;
+}
+
+
+//https://stackoverflow.com/questions/33044735/c-generating-random-numbers-inside-loop/33045918
+std::default_random_engine dre (std::chrono::steady_clock::now().time_since_epoch().count());
+int random (int lim){
+    std::uniform_int_distribution<int> uid{0, lim};
+    return uid(dre);
+}
+
+
+std::vector <int> newTasksOrder(std::vector <int> &orderedTasks){
```

Comment:
You are passing a reference so why returning same variable? 
Also white-spaces between operators.